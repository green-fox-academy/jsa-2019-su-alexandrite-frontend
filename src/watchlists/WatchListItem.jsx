import PropTypes from 'prop-types';
import React from 'react';
import {
  Text, View, Image, TouchableHighlight,
} from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import styles from './styles';
import tradeIcon from '../../assets/icons/watchList/trade.png';
import detailIcon from '../../assets/icons/watchList/detail.png';

export default function watchListItem({
  isEdit,
  isChecked,
  onSelect,
  ticker,
  currPrice,
  dailyChange,
  volumn,
}) {
  return (
    <TouchableHighlight
      style={styles.checkButton}
      onPress={onSelect}
      disabled={!isEdit}
    >
      <View style={styles.watchListItem}>
        {isEdit && <MaterialIcons name={isChecked ? 'check-box' : 'check-box-outline-blank'} />}
        <View style={styles.watchListItemColumn}>
          <Text style={[styles.watchListItemColumnText, { fontWeight: 'bold', color: '#566ed3' }]}>
            {ticker}
          </Text>
        </View>
        {isEdit ? null
          : (
            <>
              <View style={styles.watchListItemColumn}>
                <Text style={styles.watchListItemColumnText}>
                  {currPrice}
                </Text>
              </View>
              <View style={styles.watchListItemColumn}>
                {dailyChange > 0 ? (
                  <Text style={[styles.watchListItemColumnText, { color: '#21af78' }]}>
                    {`+${dailyChange}%`}
                  </Text>
                ) : (
                  <Text style={[styles.watchListItemColumnText, { color: '#d82f2f' }]}>
                    {`${dailyChange}%`}
                  </Text>
                )}
              </View>
              <View style={styles.watchListItemColumn}>
                <Text style={styles.watchListItemColumnText}>
                  {volumn}
                </Text>
              </View>
              <View style={[styles.watchListItemColumn, { flexDirection: 'row' }]}>
                <Image source={tradeIcon} />
                <Image source={detailIcon} />
              </View>
            </>
          )}
      </View>
    </TouchableHighlight>

  );
}

watchListItem.propTypes = {
  isEdit: PropTypes.bool.isRequired,
  ticker: PropTypes.string.isRequired,
  currPrice: PropTypes.number.isRequired,
  dailyChange: PropTypes.number.isRequired,
  volumn: PropTypes.string.isRequired,
  isChecked: PropTypes.bool.isRequired,
  onSelect: PropTypes.func.isRequired,
};
