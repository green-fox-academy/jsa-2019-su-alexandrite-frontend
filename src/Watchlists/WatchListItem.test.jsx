import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { useNavigation } from 'react-navigation-hooks';
import WatchListItem from './WatchListItem';

jest.mock('react-navigation-hooks');

describe('<WatchListItem />', () => {
  it('should call navigation.push when detail button is pressed', () => {
    const push = jest.fn(() => { });
    useNavigation.mockReturnValue({ push });
    const wrapper = shallow(
      <WatchListItem
        ticker="APPL"
        currPrice={0}
        dailyChange={0.5}
        volume=""
      />,
    );
    wrapper.find('TouchableHighlight')
      .first()
      .simulate('press');
    expect(useNavigation).toHaveBeenCalled();
    expect(push).toHaveBeenCalled();
  });

  it('should render properly when dailyChange is above 0', () => {
    const tree = renderer.create(
      <WatchListItem
        ticker="APPL"
        currPrice={0}
        dailyChange={0.5}
        volume=""
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render properly when dailyChange is below 0', () => {
    const tree = renderer.create(
      <WatchListItem
        ticker="APPL"
        currPrice={0}
        dailyChange={-0.5}
        volume=""
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
