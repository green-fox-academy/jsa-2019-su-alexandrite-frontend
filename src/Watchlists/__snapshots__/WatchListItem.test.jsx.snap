// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchListItem /> should render properly when dailyChange is above 0 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#666",
            "fontSize": 12,
          },
          Object {
            "color": "#566ed3",
            "fontWeight": "bold",
          },
        ]
      }
    >
      APPL
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#666",
          "fontSize": 12,
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#666",
            "fontSize": 12,
          },
          Object {
            "color": "#21af78",
          },
        ]
      }
    >
      +0.5%
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#666",
          "fontSize": 12,
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={1}
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": -10,
          "padding": 10,
        }
      }
    >
      <Image
        source={1}
        style={null}
      />
    </View>
  </View>
</View>
`;

exports[`<WatchListItem /> should render properly when dailyChange is below 0 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#666",
            "fontSize": 12,
          },
          Object {
            "color": "#566ed3",
            "fontWeight": "bold",
          },
        ]
      }
    >
      APPL
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#666",
          "fontSize": 12,
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#666",
            "fontSize": 12,
          },
          Object {
            "color": "#d82f2f",
          },
        ]
      }
    >
      -0.5%
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#666",
          "fontSize": 12,
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={1}
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": -10,
          "padding": 10,
        }
      }
    >
      <Image
        source={1}
        style={null}
      />
    </View>
  </View>
</View>
`;
